includes/Discovery.php

```diff
@@ -0,0 +1,308 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories;
+
+use Google\Web_Stories\REST_API\Stories_Controller;
+use WP_Post;
+
+/**
+ * Discovery class.
+ */
+class Discovery {
+	/**
+	 * Initialize discovery functionality.
+	 *
+	 * @return void
+	 */
+	public function init() {
+		add_action(
+			'web_stories_story_head',
+			static function () {
+				// Theme support for title-tag is implied for stories. See _wp_render_title_tag().
+				echo '<title>' . esc_html( wp_get_document_title() ) . '</title>' . "\n";
+			},
+			1
+		);
+
+		add_action( 'web_stories_story_head', [ $this, 'print_schemaorg_metadata' ] );
+		add_action( 'web_stories_story_head', [ $this, 'print_open_graph_metadata' ] );
+		add_action( 'web_stories_story_head', [ $this, 'print_twitter_metadata' ] );
+
+		// @todo Check if there's something to skip in the new version.
+		add_action( 'web_stories_story_head', 'rest_output_link_wp_head', 10, 0 );
+		add_action( 'web_stories_story_head', 'wp_resource_hints', 2 );
+		add_action( 'web_stories_story_head', 'feed_links', 2 );
+		add_action( 'web_stories_story_head', 'feed_links_extra', 3 );
+		add_action( 'web_stories_story_head', 'rsd_link' );
+		add_action( 'web_stories_story_head', 'wlwmanifest_link' );
+		add_action( 'web_stories_story_head', 'adjacent_posts_rel_link_wp_head', 10, 0 );
+		add_action( 'web_stories_story_head', 'noindex', 1 );
+		add_action( 'web_stories_story_head', 'wp_generator' );
+		add_action( 'web_stories_story_head', 'rel_canonical' );
+		add_action( 'web_stories_story_head', 'wp_shortlink_wp_head', 10, 0 );
+		add_action( 'web_stories_story_head', 'wp_site_icon', 99 );
+		add_action( 'web_stories_story_head', 'wp_oembed_add_discovery_links' );
+	}
+
+	/**
+	 * Prints the schema.org metadata on the single story template.
+	 *
+	 * @return void
+	 */
+	public function print_schemaorg_metadata() {
+		$metadata = $this->get_schemaorg_metadata();
+
+		?>
+		<script type="application/ld+json"><?php echo wp_json_encode( $metadata, JSON_UNESCAPED_UNICODE ); ?></script>
+		<?php
+	}
+
+	/**
+	 * Get schema.org metadata for the current query.
+	 *
+	 * @return array $metadata All schema.org metadata for the post.
+	 */
+	protected function get_schemaorg_metadata() {
+		$publisher = self::get_publisher_data();
+
+		$metadata = [
+			'@context'  => 'http://schema.org',
+			'publisher' => [
+				'@type' => 'Organization',
+				'name'  => $publisher['name'],
+				'logo'  => $publisher['logo'],
+			],
+		];
+
+		/**
+		 * We're expecting a post object.
+		 *
+		 * @var WP_Post $post
+		 */
+		$post = get_queried_object();
+
+		if ( $post instanceof WP_Post ) {
+			$metadata = array_merge(
+				$metadata,
+				[
+					'@type'            => 'BlogPosting',
```

Comment:
The type was discussed here - https://github.com/Yoast/wordpress-seo/pull/13446#issuecomment-547354523

I think we landed on `article` but it is unclear.

Reply:
Yep, we need to change this.

The recommendations are here: 

https://developers.google.com/search/docs/guides/enable-web-stories

It's a bit more complicated, but `Article` as the default makes the most sense.

Reply:
Changed to `Article`